aws/resource_aws_cloudwatch_event_connection.go

```diff
@@ -0,0 +1,785 @@
+									},
+									"oauth_http_parameters": {
+										Type:     schema.TypeList,
+										Required: true,
+										MaxItems: 1,
+										Elem:     connectionHttpParameters,
+									},
+									"client_parameters": {
+										Type:     schema.TypeList,
+										Optional: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"client_id": {
+													Type:     schema.TypeString,
+													Required: true,
+												},
+												"client_secret": {
+													Type:      schema.TypeString,
+													Required:  true,
+													Sensitive: true,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+						"invocation_http_parameters": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem:     connectionHttpParameters,
+						},
+					},
+				},
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsCloudWatchEventConnectionCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudwatcheventsconn
+
+	input := &events.CreateConnectionInput{}
+
+	if name, ok := d.GetOk("name"); ok {
+		input.Name = aws.String(name.(string))
+	}
+	if description, ok := d.GetOk("description"); ok {
+		input.Description = aws.String(description.(string))
+	}
+	if authorizationType, ok := d.GetOk("authorization_type"); ok {
+		input.AuthorizationType = aws.String(authorizationType.(string))
+	}
+	if authParameters, ok := d.GetOk("auth_parameters"); ok {
+		input.AuthParameters = expandAwsCloudWatchEventCreateConnectionAuthRequestParameters(authParameters.([]interface{}))
+	}
+
+	log.Printf("[DEBUG] Creating CloudWatchEvent connection: %v", input)
+
+	_, err := conn.CreateConnection(input)
+	if err != nil {
+		return fmt.Errorf("Creating CloudWatchEvent connection (%s) failed: %w", *input.Name, err)
+	}
+
+	d.SetId(aws.StringValue(input.Name))
+
+	log.Printf("[INFO] CloudWatchEvent connection (%s) created", d.Id())
+
+	return resourceAwsCloudWatchEventConnectionRead(d, meta)
+}
+
+func resourceAwsCloudWatchEventConnectionRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudwatcheventsconn
+
+	input := &events.DescribeConnectionInput{
+		Name: aws.String(d.Id()),
+	}
+
+	log.Printf("[DEBUG] Reading CloudWatchEvent connection (%s)", d.Id())
+	output, err := conn.DescribeConnection(input)
+	if isAWSErr(err, events.ErrCodeResourceNotFoundException, "") {
+		log.Printf("[WARN] CloudWatchEvent connection (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("error reading CloudWatchEvent connection: %w", err)
+	}
+
+	log.Printf("[DEBUG] Found CloudWatchEvent connection: %#v", *output)
+
+	d.Set("arn", output.ConnectionArn)
```

Comment:
Please also include `secret_arn`: https://docs.aws.amazon.com/sdk-for-go/api/service/cloudwatchevents/#CreateConnectionOutput